server/services/v1/search_reader.go

```diff
@@ -104,13 +103,20 @@ func (p *pageReader) read() error {
 	}
 
 	sortedHits := tsearch.NewSortedHits(p.query.SortOrder)
+	sortedFacets := tsearch.NewSortedFacets()
 	for _, r := range result {
 		if r.Hits != nil {
 			for _, h := range *r.Hits {
 				hit := tsearch.NewSearchHit(&h)
-				err := sortedHits.Add(hit)
-				// log and skip hits that cannot be added
-				if ulog.E(err) {
+				if ulog.E(sortedHits.Add(hit)) {
+					continue
+				}
+			}
+		}
+
+		if r.FacetCounts != nil {
+			for _, fc := range *r.FacetCounts {
+				if ulog.E(sortedFacets.Add(&fc)) {
```

Comment:
store the facets from multiple results in `SortedFacets` that will remove duplicates and arrange facets in descending order